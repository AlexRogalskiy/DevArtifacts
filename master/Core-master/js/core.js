/**
 * Copyright (C) 2011 Hakim El Hattab, http://hakim.se
 * 
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * 
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 * 
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 * 
 * 
 * @author Hakim El Hattab | http://hakim.se
 */
var Core = new function(){
	
	// Flags if we are running mobile mode
	var isMobile = !!navigator.userAgent.toLowerCase().match( /ipod|ipad|iphone|android/gi );
	
	var DEFAULT_WIDTH = 900,
		DEFAULT_HEIGHT = 550,
		BORDER_WIDTH = 6,
		FRAMERATE = 60;
	
	// Types of organisms
	var ORGANISM_ENEMY = 'enemy',
		ORGANISM_ENERGY = 'energy';
	
	// The world dimensions
	var world = {
		width: isMobile ? window.innerWidth : DEFAULT_WIDTH,
		height: isMobile ? window.innerHeight : DEFAULT_HEIGHT
	};
	
	var canvas,
		context;
	
	var canvasBackground,
		contextBackground;
	
	// UI DOM elements
	var status;
	var panels;
	var message;
	var title;
	var startButton;
	
	// Game elements
	var organisms = [];
	var particles = [];
	var player;
	
	// Mouse properties
	var mouseX = (window.innerWidth + world.width) * 0.5;
	var mouseY = (window.innerHeight + world.height) * 0.5;
	var mouseIsDown = false;
	var spaceIsDown = false;
	
	// Game properties and scoring
	var playing = false;
	var score = 0;
	var time = 0;
	var duration = 0;
	var difficulty = 1;
	var lastspawn = 0;
	
	// Game statistics
	var fc = 0; // Frame count
	var fs = 0; // Frame score
	var cs = 0; // Collision score
	
	// The world's velocity
	var velocity = { x: -1.3, y: 1 };
	
	// Performance (FPS) tracking
	var fps = 0;
	var fpsMin = 1000;
	var fpsMax = 0;
	var timeLastSecond = new Date().getTime();
	var frames = 0;
	
	this.init = function(){
	
		canvas = document.getElementById('world');
		canvasBackground = document.getElementById('background');
		panels = document.getElementById('panels');
		status = document.getElementById('status');
		message = document.getElementById('message');
		title = document.getElementById('title');
		startButton = document.getElementById('startButton');
		
		if (canvas && canvas.getContext) {
			context = canvas.getContext('2d');
			
			contextBackground = canvasBackground.getContext('2d');
			
			// Register event listeners
			document.addEventListener('mousemove', documentMouseMoveHandler, false);
			document.addEventListener('mousedown', documentMouseDownHandler, false);
			document.addEventListener('mouseup', documentMouseUpHandler, false);
			canvas.addEventListener('touchstart', documentTouchStartHandler, false);
			document.addEventListener('touchmove', documentTouchMoveHandler, false);
			document.addEventListener('touchend', documentTouchEndHandler, false);
			window.addEventListener('resize', windowResizeHandler, false);
			startButton.addEventListener('click', startButtonClickHandler, false);
			document.addEventListener('keydown', documentKeyDownHandler, false);
			document.addEventListener('keyup', documentKeyUpHandler, false);
			
			// Define our player
			player = new Player();
			
			// Force an initial resize to make sure the UI is sized correctly
			windowResizeHandler();
			
			// If we are running on mobile, certain elements need to be configured differently
			if( isMobile ) {
				document.getElementsByTagName('header')[0].style.display = 'none';
				status.style.width = world.width + 'px';
				canvas.style.border = 'none';
			}
			
			animate();
		}
	};
	
	function renderBackground() {
		var gradient = contextBackground.createRadialGradient( world.width * 0.5, world.height * 0.5, 0, world.width * 0.5, world.height * 0.5, 500 );
		gradient.addColorStop(0,'rgba(0, 70, 70, 1)');
		gradient.addColorStop(1,'rgba(0, 8, 14, 1)');
		
		contextBackground.fillStyle = gradient;
		contextBackground.fillRect( 0, 0, world.width, world.height );
	}
	
	/**
	 * Handles click on the start button in the UI.
	 */
	function startButtonClickHandler(event){
		if( playing == false ) {
			playing = true;
			
			// Reset game properties
			organisms = [];
			score = 0;
			difficulty = 1;
			
			// Reset game tracking properties
			fc = 0;
			fs = 0;
			ms = 0;
			cs = 0;
			
			// Reset the player data
			player.energy = 30;
			
			// Hide the game UI
			panels.style.display = 'none';
			status.style.display = 'block';
			
			time = new Date().getTime();
		}
	}
	
	/**
	 * Stops the currently ongoing game and shows the
	 * resulting data in the UI.
	 */
	function gameOver() {
		playing = false;
		
		// Determine the duration of the game
		duration = new Date().getTime() - time;
		
		// Show the UI
		panels.style.display = 'block';
		
		// Ensure that the score is an integer
		score = Math.round(score);
		
		// Write the users score to the UI
		title.innerHTML = 'Game Over! (' + score + ' points)';
		
		// Update the status bar with the final score and time
		scoreText = 'Score: <span>' + Math.round( score ) + '</span>';
		scoreText += ' Time: <span>' + Math.round( ( ( new Date().getTime() - time ) / 1000 ) * 100 ) / 100 + 's</span>';
		status.innerHTML = scoreText;
	}
	
	function documentKeyDownHandler(event) {
		switch( event.keyCode ) {
			case 32:
				if( !spaceIsDown && player.energy > 15 ) {
					player.energy -= 4;
				}
				spaceIsDown = true;
				event.preventDefault();
				break;
		}
	}
	function documentKeyUpHandler(event) {
		switch( event.keyCode ) {
			case 32:
				spaceIsDown = false;
				event.preventDefault();
				break;
		}
	}
	
	/**
	 * Event handler for document.onmousemove.
	 */
	function documentMouseMoveHandler(event){
		mouseX = event.clientX - (window.innerWidth - world.width) * 0.5 - BORDER_WIDTH;
		mouseY = event.clientY - (window.innerHeight - world.height) * 0.5 - BORDER_WIDTH;
	}
	
	/**
	 * Event handler for document.onmousedown.
	 */
	function documentMouseDownHandler(event){
		mouseIsDown = true;
	}
	
	/**
	 * Event handler for document.onmouseup.
	 */
	function documentMouseUpHandler(event) {
		mouseIsDown = false;
	}
	
	/**
	 * Event handler for document.ontouchstart.
	 */
	function documentTouchStartHandler(event) {
		if(event.touches.length == 1) {
			event.preventDefault();
			
			mouseX = event.touches[0].pageX - (window.innerWidth - world.width) * 0.5;
			mouseY = event.touches[0].pageY - (window.innerHeight - world.height) * 0.5;
			
			mouseIsDown = true;
		}
	}
	
	/**
	 * Event handler for document.ontouchmove.
	 */
	function documentTouchMoveHandler(event) {
		if(event.touches.length == 1) {
			event.preventDefault();

			mouseX = event.touches[0].pageX - (window.innerWidth - world.width) * 0.5 - 60;
			mouseY = event.touches[0].pageY - (window.innerHeight - world.height) * 0.5 - 30;
		}
	}
	
	/**
	 * Event handler for document.ontouchend.
	 */
	function documentTouchEndHandler(event) {
		mouseIsDown = false;
	}
	
	/**
	 * Event handler for window.onresize.
	 */
	function windowResizeHandler() {
		// Update the game size
		world.width = isMobile ? window.innerWidth : DEFAULT_WIDTH;
		world.height = isMobile ? window.innerHeight : DEFAULT_HEIGHT;
		
		// Center the player
		player.position.x = world.width * 0.5;
		player.position.y = world.height * 0.5;
		
		// Resize the canvas
		canvas.width = world.width;
		canvas.height = world.height;
		canvasBackground.width = world.width;
		canvasBackground.height = world.height;
		
		// Determine the x/y position of the canvas
		var cvx = (window.innerWidth - world.width) * 0.5;
		var cvy = (window.innerHeight - world.height) * 0.5;
		
		// Position the canvas
		canvas.style.position = 'absolute';
		canvas.style.left = cvx + 'px';
		canvas.style.top = cvy + 'px';
		canvasBackground.style.position = 'absolute';
		canvasBackground.style.left = cvx + BORDER_WIDTH + 'px';
		canvasBackground.style.top = cvy + BORDER_WIDTH + 'px';
		
		if( isMobile ) {
			panels.style.left = '0px';
			panels.style.top = '0px';
			status.style.left = '0px';
			status.style.top = '0px';
		}
		else {
			panels.style.left = cvx + BORDER_WIDTH + 'px';
			panels.style.top = cvy + 200 + 'px';
			status.style.left = cvx + BORDER_WIDTH + 'px';
			status.style.top = cvy + BORDER_WIDTH + 'px';
		}
		
		renderBackground();
	}
	
	/**
	 * Emits a random number of particles from a set point.
	 * 
	 * @param position The point where particles will be 
	 * emitted from
	 * @param spread The pixel spread of the emittal
	 */
	function emitParticles( position, direction, spread, seed ) {
		var q = seed + ( Math.random() * seed );
		
		while( --q >= 0 ) {
			var p = new Point();
			p.position.x = position.x + ( Math.sin(q) * spread );
			p.position.y = position.y + ( Math.cos(q) * spread );
			p.velocity = { x: direction.x + ( -1 + Math.random() * 2 ), y: direction.y + ( - 1 + Math.random() * 2 ) };
			p.alpha = 1;
			
			particles.push( p );
		}
	}
	
	/**
	 * Called on every frame to update the game properties
	 * and render the current state to the canvas.
	 */
	function animate() {
		
		// Fetch the current time for this frame
		var frameTime = new Date().getTime();
		
		// Increase the frame count
		frames ++;
		
		// Check if a second has passed since the last time we updated the FPS
		if( frameTime > timeLastSecond + 1000 ) {
			// Establish the current, minimum and maximum FPS
			fps = Math.min( Math.round( ( frames * 1000 ) / ( frameTime - timeLastSecond ) ), FRAMERATE );
			fpsMin = Math.min( fpsMin, fps );
			fpsMax = Math.max( fpsMax, fps );
			
			timeLastSecond = frameTime;
			frames = 0;
		}
		
		// A factor by which the score will be scaled, depending on current FPS
		var scoreFactor = 0.01 + ( Math.max( Math.min( fps, FRAMERATE ), 0 ) / FRAMERATE * 0.99 );
		
		// Scales down the factor by itself
		scoreFactor = scoreFactor * scoreFactor;
		
		// Clear the canvas of all old pixel data
		context.clearRect(0,0,canvas.width,canvas.height);
		
		var i, 
			ilen, 
			j, 
			jlen;
		
		// Only update game properties and draw the player if a game is active
		if( playing ) {
			
			// Increment the difficulty slightly
			difficulty += 0.0015;
			
			// Increment the score depending on difficulty
			score += (0.4 * difficulty) * scoreFactor;
			
			// Increase the game frame count stat
			fc ++;
			
			// Increase the score count stats
			fs += (0.4 * difficulty) * scoreFactor;
			
			//player.angle = Math.atan2( mouseY - player.position.y, mouseX - player.position.x );
			
			var targetAngle = Math.atan2( mouseY - player.position.y, mouseX - player.position.x );
			
			if( Math.abs( targetAngle - player.angle ) > Math.PI ) {
				player.angle = targetAngle;
			}
			
			player.angle += ( targetAngle - player.angle ) * 0.2;
			
			player.energyRadiusTarget = ( player.energy / 100 ) * ( player.radius * 0.8 );
			player.energyRadius += ( player.energyRadiusTarget - player.energyRadius ) * 0.2;
			
			player.shield = { x: player.position.x + Math.cos( player.angle ) * player.radius, y: player.position.y + Math.sin( player.angle ) * player.radius };
			
			// Shield
			context.beginPath();
			context.strokeStyle = '#648d93';
			context.lineWidth = 3;
			context.arc( player.position.x, player.position.y, player.radius, player.angle + 1.6, player.angle - 1.6, true );
			context.stroke();
			
			// Core
			context.beginPath();
			context.fillStyle = "#249d93";
			context.strokeStyle = "#3be2d4";
			context.lineWidth = 1.5;
			
			player.updateCore();
			
			var loopedNodes = player.coreNodes.concat();
			loopedNodes.push( player.coreNodes[0] );
			
			for( var i = 0; i < loopedNodes.length; i++ ) {
				p = loopedNodes[i];
				p2 = loopedNodes[i+1];
				
				p.position.x += ( (player.position.x + p.normal.x + p.offset.x) - p.position.x ) * 0.2;
				p.position.y += ( (player.position.y + p.normal.y + p.offset.y) - p.position.y ) * 0.2;
				
				if( i == 0 ) {
					// This is the first loop, so we need to start by moving into position
					context.moveTo( p.position.x, p.position.y );
				}
				else if( p2 ) {
					// Draw a curve between the current and next trail point
					context.quadraticCurveTo( p.position.x, p.position.y, p.position.x + ( p2.position.x - p.position.x ) / 2, p.position.y + ( p2.position.y - p.position.y ) / 2 );
				}
			}
			
			context.closePath();
			context.fill();
			context.stroke();
			
		}
		
		if( spaceIsDown && player.energy > 10 ) {
			player.energy -= 0.1;
			
			context.beginPath();
			context.fillStyle = 'rgba( 0, 100, 100, ' + ( player.energy / 100 ) * 0.9 + ' )';
			context.arc( player.position.x, player.position.y, player.radius, 0, Math.PI*2, true );
			context.fill();
		}
		
		var enemyCount = 0;
		var energyCount = 0;
		
		// Go through each enemy and draw it + update its properties
		for( i = 0; i < organisms.length; i++ ) {
			p = organisms[i];
			
			p.position.x += p.velocity.x;
			p.position.y += p.velocity.y;
			
			p.alpha += ( 1 - p.alpha ) * 0.1;
			
			if( p.type == ORGANISM_ENEMY ) context.fillStyle = 'rgba( 255, 0, 0, ' + p.alpha + ' )';
			if( p.type == ORGANISM_ENERGY ) context.fillStyle = 'rgba( 0, 235, 190, ' + p.alpha + ' )';
			
			context.beginPath();
			context.arc(p.position.x, p.position.y, p.size/2, 0, Math.PI*2, true);
			context.fill();
			
			var angle = Math.atan2( p.position.y - player.position.y, p.position.x - player.position.x );
			
			if (playing) {
				
				var dist = Math.abs( angle - player.angle );
				
				if( dist > Math.PI ) {
					dist = ( Math.PI * 2 ) - dist;
				}
				
				if ( dist < 1.6 ) {
					if (p.distanceTo(player.position) > player.radius - 5 && p.distanceTo(player.position) < player.radius + 5) {
						p.dead = true;
					}
				}
				
				if (spaceIsDown && p.distanceTo(player.position) < player.radius && player.energy > 11) {
					p.dead = true;
					score += 4;
				}
				
				if (p.distanceTo(player.position) < player.energyRadius + (p.size * 0.5)) {
					if (p.type == ORGANISM_ENEMY) {
						player.energy -= 6;
					}
					
					if (p.type == ORGANISM_ENERGY) {
						player.energy += 8;
						score += 30;
					}
					
					player.energy = Math.max(Math.min(player.energy, 100), 0);
					
					p.dead = true;
				}
			}
			
			// If the enemy is outside of the game bounds, destroy it
			if( p.position.x < -p.size || p.position.x > world.width + p.size || p.position.y < -p.size || p.position.y > world.height + p.size ) {
				p.dead = true;
			}
			
			// If the enemy is dead, remove it
			if( p.dead ) {
				emitParticles( p.position, { x: (p.position.x - player.position.x) * 0.02, y: (p.position.y - player.position.y) * 0.02 }, 5, 5 );
				
				organisms.splice( i, 1 );
				i --;
			}
			else {
				if( p.type == ORGANISM_ENEMY ) enemyCount ++;
				if( p.type == ORGANISM_ENERGY ) energyCount ++;
			}
		}
		
		// If there are less enemies than intended for this difficulty, add another one
		if( enemyCount < 1 * difficulty && new Date().getTime() - lastspawn > 100 ) {
			organisms.push( giveLife( new Enemy() ) );
			lastspawn = new Date().getTime();
		}
		
		// 
		if( energyCount < 1 && Math.random() > 0.996 ) {
			organisms.push( giveLife( new Energy() ) );
		}
		
		// Go through and draw all particle effects
		for( i = 0; i < particles.length; i++ ) {
			p = particles[i];
			
			// Apply velocity to the particle
			p.position.x += p.velocity.x;
			p.position.y += p.velocity.y;
			
			// Fade out
			p.alpha -= 0.02;
			
			// Draw the particle
			context.fillStyle = 'rgba(255,255,255,'+Math.max(p.alpha,0)+')';
			context.fillRect( p.position.x, p.position.y, 1, 1 );
			
			// If the particle is faded out to less than zero, remove it
			if( p.alpha <= 0 ) {
				particles.splice( i, 1 );
			}
		}
		
		// If the game is active, update the game status bar with score, duration and FPS
		if( playing ) {
			scoreText = 'Score: <span>' + Math.round( score ) + '</span>';
			scoreText += ' Time: <span>' + Math.round( ( ( new Date().getTime() - time ) / 1000 ) * 100 ) / 100 + 's</span>';
			scoreText += ' <p class="fps">FPS: <span>' + Math.round( fps ) + ' ('+Math.round(Math.max(Math.min(fps/FRAMERATE, FRAMERATE), 0)*100)+'%)</span></p>';
			status.innerHTML = scoreText;
			
			if( player.energy === 0 ) {
				emitParticles( player.position, { x: 0, y: 0 }, 10, 40 );
				
				gameOver();
			}
		}
		
		requestAnimFrame( animate );
	}
	
	/**
	 * 
	 */
	function giveLife( organism ) {
		var side = Math.round( Math.random() * 3 );
		
		switch( side ) {
			case 0:
				organism.position.x = 10;
				organism.position.y = world.height * Math.random();
				break;
			case 1:
				organism.position.x = world.width * Math.random();
				organism.position.y = 10;
				break;
			case 2:
				organism.position.x = world.width - 10;
				organism.position.y = world.height * Math.random();
				break;
			case 3:
				organism.position.x = world.width * Math.random();
				organism.position.y = world.height - 10;
				break;
		}
		
		organism.speed = Math.min( Math.max( Math.random(), 0.6 ), 0.75 );
		
		organism.velocity.x = ( player.position.x - organism.position.x ) * 0.006 * organism.speed;
		organism.velocity.y = ( player.position.y - organism.position.y ) * 0.006 * organism.speed;
		
		if( organism.type == 'enemy' ) {
			organism.velocity.x *= (1+(Math.random()*0.1));
			organism.velocity.y *= (1+(Math.random()*0.1));
		}
		
		organism.alpha = 0;
		
		return organism;
	}
	
};

function Point( x, y ) {
	this.position = { x: x, y: y };
}
Point.prototype.distanceTo = function(p) {
	var dx = p.x-this.position.x;
	var dy = p.y-this.position.y;
	return Math.sqrt(dx*dx + dy*dy);
};
Point.prototype.clonePosition = function() {
	return { x: this.position.x, y: this.position.y };
};

function Player() {
	this.position = { x: 0, y: 0 };
	this.length = 15;
	this.energy = 30;
	this.energyRadius = 0;
	this.energyRadiusTarget = 0;
	this.radius = 60;
	this.angle = 0;
	this.coreQuality = 16;
	this.coreNodes = [];
}
Player.prototype = new Point();
Player.prototype.updateCore = function() {
	var i, j, n;
	
	if( this.coreNodes.length == 0 ) {
		var i, n;
		
		for (i = 0; i < this.coreQuality; i++) {
			n = {
				position: { x: this.position.x, y: this.position.y },
				normal: { x: 0, y: 0 },
				normalTarget: { x: 0, y: 0 },
				offset: { x: 0, y: 0 }
			};
			
			this.coreNodes.push( n );
		}
	}
	
	for (i = 0; i < this.coreQuality; i++) {
		
		var n = this.coreNodes[i];
		
		var angle = ( i / this.coreQuality ) * Math.PI * 2;
		
		n.normal.x = Math.cos( angle ) * this.energyRadius;
		n.normal.y = Math.sin( angle ) * this.energyRadius;
		
		n.offset.x = Math.random() * 5;
		n.offset.y = Math.random() * 5;
	}
};

function Enemy() {
	this.position = { x: 0, y: 0 };
	this.velocity = { x: 0, y: 0 };
	this.size = 6 + ( Math.random() * 4 );
	this.speed = 1;
	this.type = 'enemy';
}
Enemy.prototype = new Point();

function Energy() {
	this.position = { x: 0, y: 0 };
	this.velocity = { x: 0, y: 0 };
	this.size = 10 + ( Math.random() * 6 );
	this.speed = 1;
	this.type = 'energy';
}
Energy.prototype = new Point();

// shim with setTimeout fallback from http://paulirish.com/2011/requestanimationframe-for-smart-animating/
window.requestAnimFrame = (function(){
  return  window.requestAnimationFrame       || 
          window.webkitRequestAnimationFrame || 
          window.mozRequestAnimationFrame    || 
          window.oRequestAnimationFrame      || 
          window.msRequestAnimationFrame     || 
          function(/* function */ callback, /* DOMElement */ element){
            window.setTimeout(callback, 1000 / 60);
          };
})();

Core.init();