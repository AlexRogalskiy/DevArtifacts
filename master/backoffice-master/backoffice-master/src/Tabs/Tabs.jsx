import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'
import {
  AppBar,
  withStyles,
  Tabs as MaterialTabs,
  Tab,
  Typography,
} from '@material-ui/core'

function TabContainer(props) {
  return (
    <Typography component="div" style={{ padding: 8 * 3 }}>
      {props.children}
    </Typography>
  )
}

TabContainer.propTypes = {
  children: PropTypes.node.isRequired,
}

const styles = theme => ({
  root: {
    flexGrow: 1,
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    backgroundColor: theme.palette.background.paper,
  },
})

class Tabs extends React.Component {
  static propTypes = {
    isScrollable: PropTypes.bool,
    data: PropTypes.arrayOf(PropTypes.object).isRequired,
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
  }

  static defaultProps = {
    isScrollable: false,
  }

  constructor() {
    super()

    this.handleChange = this.handleChange.bind(this)
  }

  state = {
    value: 0,
  }

  componentWillMount() {
    const { data } = this.props
    const hash = window.location.hash.replace('#/', '')
    let value = 0

    if (data && data.constructor === Array) {
      data.forEach((item, index) => {
        if (item.id === hash) {
          value = index
        }
      })
    }

    this.setState({
      value,
    })
  }

  handleChange(event, value) {
    const hash = this.props.data[value].id || value

    window.location.hash = `#/${hash}`

    this.setState({ value })
  }

  render() {
    const { isScrollable, data, classes } = this.props
    const { value } = this.state

    return (
      <div className={classes.root}>
        <AppBar position="static">
          <MaterialTabs
            value={value}
            onChange={this.handleChange}
            indicatorColor="secondary"
            scrollable={isScrollable}
            scrollButtons="auto"
          >
            {data.map(item => (
              <Tab
                key={`tab-${(Math.random() * 10000).toFixed(4)}`}
                label={item.title}
              />
            ))}
          </MaterialTabs>
        </AppBar>

        {data.map((item, index) => {
          if (value === index) {
            return (
              <TabContainer
                key={`tabcontainer-${(Math.random() * 10000).toFixed(4)}`}
              >
                {item.content}
              </TabContainer>
            )
          }

          return null
        })}
      </div>
    )
  }
}

const tabsWithStyles = withStyles(styles)(Tabs)
export default withRouter(tabsWithStyles)
