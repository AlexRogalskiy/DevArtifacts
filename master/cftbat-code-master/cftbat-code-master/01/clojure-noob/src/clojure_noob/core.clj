(ns clojure-noob.core
  (:gen-class))

(defn -main
  "I don't do a whole lot ... yet."
  [& args]
  (println "I'm a little teapot!"))
