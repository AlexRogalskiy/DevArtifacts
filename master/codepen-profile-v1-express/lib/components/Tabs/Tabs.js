/**
 * External dependencies
 */
import React, { PropTypes } from 'react';
import { Link } from 'react-router';

/**
 * Local variables
 */
const propTypes = {
  params: PropTypes.object,
};

const defaultProps = {};

/**
 * <Tabs />
 */
const Tabs = ({ params }) => (
  <nav className="tabs">
    <Link
      activeClassName="tabs__item--active"
      className="tabs__item"
      to={`/${params.view}/public`}
    >
      Public
    </Link>
    <Link
      activeClassName="tabs__item--active"
      className="tabs__item"
      to={`/${params.view}/popular`}
    >
      Popular
    </Link>
    <Link
      activeClassName="tabs__item--active"
      className="tabs__item"
      to={`/${params.view}/loved`}
    >
      Loved
    </Link>
    <Link
      activeClassName="tabs__item--active"
      className="tabs__item"
      to={`/${params.view}/forked`}
    >
      Forked
    </Link>
  </nav>
);

Tabs.propTypes = propTypes;
Tabs.defaultProps = defaultProps;

export default Tabs;
