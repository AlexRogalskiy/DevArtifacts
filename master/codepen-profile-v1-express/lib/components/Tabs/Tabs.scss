// Tabs
.tabs {
  &__item {
    position: relative;
    top: 1px;
    display: inline-block;
    margin: 0 32px 0 0;
    color: $text-secondary;
    font-size: 0.875rem;
    line-height: 40px;
    text-decoration: none;

    &:hover {
      border-bottom: 1px solid rgba($text-primary, 0.5);
      color: $text-primary;
    }

    a {
      color: inherit;
    }

    &--active {
      border-bottom: 1px solid $text-primary;
      color: $text-primary;
      font-weight: $font-semibold;
    }
  }
}
