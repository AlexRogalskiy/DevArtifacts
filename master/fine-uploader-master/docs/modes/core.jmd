{% extends "_templates/base.html" %}
{% set page_title = "Core Mode" %}
{% block content %}
{% markdown %}
# Fine Uploader Core Mode {: .page-header }

[solo-dnd]: ../features/drag-and-drop.html#drag-and-drop-standalone-module
[ui]: ui.html

If you need to design an exceptionally unique upload experience for you users, _and_ require the
power that Fine Uploader provides in its feature set, core mode is most appropriate. While [UI mode][ui]
provides a customizable default UI, core mode is designed to allow you to develop your own UI that builds on
Fine Uploader's rich API and configuration options. There is _also_ a [standalone drag and drop module][solo-dnd]
available for you to integrate into your custom UI.

Fine Uploader Core mode is defined in the `qq.FineUploaderBasic` module in the code.
This is the base module for Fine Uploader, and provides all functions that do not 
involve UI components.

For more information on the events, API methods, and configuration options available to UI mode,
have a look at the menu at the top of the page.

{% endmarkdown %}
{% endblock %}
