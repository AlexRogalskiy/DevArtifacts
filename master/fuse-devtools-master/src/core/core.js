var CoreAPI = (function() {

    'use strict';
    
    function getGreeting() {
        return 'Hello from the Core API';
    }

    return {
        getGreeting: getGreeting
    };

}());

// Expose the CoreAPI for node unit testing
if (typeof module !== 'undefined') {
    module.exports = CoreAPI;
}
