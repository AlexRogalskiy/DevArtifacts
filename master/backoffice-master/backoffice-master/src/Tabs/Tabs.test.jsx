import React from 'react'
import MockRouter from 'react-mock-router'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Tab } from '@material-ui/core'

import Tabs from './'
import tabsContent from '../../__tests__/data/tabs'

Enzyme.configure({ adapter: new Adapter() })

describe('Tabs', () => {
  it('renders correctly', () => {
    const tree = mount((
      <MockRouter>
        <Tabs
          data={[{
            title: 'Title',
            content: (
              <p>Content</p>
            ),
          }]}
        />
      </MockRouter>
    ))

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly if no data', () => {
    const tree = mount((
      <MockRouter>
        <Tabs data={[]} />
      </MockRouter>
    ))

    expect(tree).toMatchSnapshot()
  })

  it('click second tab', () => {
    const tabAt = 1
    const tabs = mount(<MockRouter><Tabs data={tabsContent} /></MockRouter>)

    expect(window.location.hash).toBe('')
    tabs.find(Tab).at(tabAt).simulate('click')
    expect(window.location.hash).toBe(`#/${tabsContent[tabAt].id}`)
  })

  it('test if correct element is active', () => {
    const tabAt = 1
    window.location.hash = `#/${tabsContent[tabAt].id}`

    mount(<MockRouter><Tabs data={tabsContent} /></MockRouter>)

    expect(window.location.hash).toBe(`#/${tabsContent[tabAt].id}`)
  })
})
