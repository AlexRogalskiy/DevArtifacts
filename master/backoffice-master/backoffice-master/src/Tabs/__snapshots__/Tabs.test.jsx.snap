// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
<div
  className="Tabs-root-1"
>
  <header
    className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-8"
  >
    <div
      className="MuiTabs-root-37"
    >
      <div
        className="MuiTabs-flexContainer-38"
      >
        <div
          className="MuiTabs-scroller-40 MuiTabs-fixed-41"
          onScroll={[Function]}
          role="tablist"
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <div
            className="MuiTabs-flexContainer-38"
          >
            <button
              aria-selected={true}
              className="MuiButtonBase-root-58 MuiTab-root-46 MuiTab-textColorInherit-48 MuiTab-selected-51"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiTab-wrapper-54"
              >
                <span
                  className="MuiTab-labelContainer-55"
                >
                  <span
                    className="MuiTab-label-56"
                  >
                    Title
                  </span>
                </span>
              </span>
              <span
                className="MuiTouchRipple-root-64"
              />
            </button>
          </div>
          <span
            className="TabIndicator-root-61 TabIndicator-colorSecondary-63 MuiTabs-indicator-45"
            style={
              Object {
                "left": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
  </header>
  <div
    className="MuiTypography-root-71 MuiTypography-body1-80"
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <p>
      Content
    </p>
  </div>
</div>
`;

exports[`Tabs renders correctly if no data 1`] = `
<div
  className="Tabs-root-1"
>
  <header
    className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-8"
  >
    <div
      className="MuiTabs-root-37"
    >
      <div
        className="MuiTabs-flexContainer-38"
      >
        <div
          className="MuiTabs-scroller-40 MuiTabs-fixed-41"
          onScroll={[Function]}
          role="tablist"
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <div
            className="MuiTabs-flexContainer-38"
          />
          <span
            className="TabIndicator-root-61 TabIndicator-colorSecondary-63 MuiTabs-indicator-45"
            style={
              Object {
                "left": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
